### PR TITLE
Add a tool to determinize prow configuration

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: cmd/determinize-prow-config/main.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.prowConfigDir, "prow-config-dir", "", "Path to the Prow configuration directory.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
```
Comment (droslean) [inline cmd/determinize-prow-config/main.go]: we don't want to continue if that happens, right?

File: cmd/determinize-prow-config/main.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		logrus.WithError(err).Fatal("could not update Prow configuration")
+	}
+
+	if err := updatePluginConfig(o.prowConfigDir); err != nil {
+		logrus.WithError(err).Fatal("could not update Prow plugin configuration")
+	}
+}
+
+func updateProwConfig(configDir string) error {
+	configPath := path.Join(configDir, "_config.yaml")
```
Comment (droslean) [inline cmd/determinize-prow-config/main.go]: lets put this in a const

File: cmd/determinize-prow-config/main.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	data, err := yaml.Marshal(agent.Config())
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(configPath, data, 0644)
+}
+
+func updatePluginConfig(configDir string) error {
+	configPath := path.Join(configDir, "_plugins.yaml")
```
Comment (droslean) [inline cmd/determinize-prow-config/main.go]: this too.

File: cmd/determinize-prow-config/main.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	agent := prowconfig.Agent{}
+	if err := agent.Start(configPath, ""); err != nil {
+		return fmt.Errorf("could not load Prow configuration: %v", err)
+	}
+	data, err := yaml.Marshal(agent.Config())
+	if err != nil {
+		return fmt.Errorf("could not marshal Prow configuration: %v", err)
+	}
+
+	return ioutil.WriteFile(configPath, data, 0644)
```
Comment (droslean) [inline cmd/determinize-prow-config/main.go]: Why do we need to update the existing file?

Comment (stevekuznetsov) [reply cmd/determinize-prow-config/main.go]: That's the whole point of the tool?

File: cmd/determinize-prow-config/main.go (hunk 1 above)
Comment (stevekuznetsov) [reply cmd/determinize-prow-config/main.go]: `Fatal` will kill it, yes

File: cmd/determinize-prow-config/main.go (hunk 4 above)
Comment (droslean) [reply cmd/determinize-prow-config/main.go]: Sry, I had a different meaning of `determinize` in my mind :)

File: cmd/determinize-prow-config/main.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
+// determinize-prow-config reads and writes Prow configuration
+// to enforce formatting on the files
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/openshift/ci-tools/pkg/config"
```
Comment (hongkailiu) [inline cmd/determinize-prow-config/main.go]: NIP: drag down?

**Pull request merged**